### PR TITLE
feat: add basic tools for app real-time monitoring

### PR DESCRIPTION
**Pull request by micieslak**
Add configuration and basic C++ class allowing run-time inspection into the application.

Required by: https://github.com/status-im/status-desktop/issues/8786

### CODE REVIEW

File: lib/src/Status/Monitoring/Monitor.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                     [engine](QObject *obj, const QUrl &objUrl) {
+        if (!obj) {
+            qWarning() << "Error while loading QML:" << objUrl << "."
+                       << "Monitor initialization failed.";
+            return;
+        }
+
+        QQuickWindow* window = qobject_cast<QQuickWindow*>(obj);
+        QQmlComponent cmp(engine, QCoreApplication::applicationDirPath()
+                          + QStringLiteral("/../monitoring/Main.qml"), window);
```
Comment (osmaczko) [inline lib/src/Status/Monitoring/Monitor.cpp:20]: That looks like the wrong dependency direction. Would be great if DOtherSide doesn't make any assumptions about clients. In this case, it requires clients to have `"/../monitoring/Main.qml"`.

What would you say about passing the entry point through the definition provided by clients?
```suggestion
        QQmlComponent cmp(engine, QCoreApplication::applicationDirPath()
                          + MONITORING_ENTRY_POINT, window);
```

**Review by osmaczko — COMMENTED** [review]
I like the approach, looks non-invasive :+1:

File: lib/src/Status/Monitoring/Monitor.cpp (hunk 1 above)
Comment (micieslak) [reply lib/src/Status/Monitoring/Monitor.cpp:20]: Yeah, I know, dependency direction is wrong, I didn't feel comfortable about that but just didn't have any better idea. But yeah, I like your proposition, I will try to pass it from Makefile, through CMakeLists.txt to the source. Thanks!

**Review by alexjba — APPROVED** [review]
Nice work! 👍

**Review by osmaczko — APPROVED** [review]

**Review by alexandraB99 — APPROVED** [review]

**Pull request merged**